### PR TITLE
Add ElContracts Writer tests

### PR DESCRIPTION
**Pull request by pablodeymo**
### What Changed?
- Added tests for ElContracts Writer
- Moved `BuildTestClients` function into a new testutils subpackage 

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

File: chainio/clients/builder.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type BuildAllConfig struct {
 	OperatorStateRetrieverAddr string
 	AvsName                    string
 	PromMetricsIpPortAddress   string
+	RewardsCoordinatorAddress  string
```
Comment (afkbyte) [inline chainio/clients/builder.go]: why add the `RewardsCoordinator` here?

File: chainio/clients/elcontracts/writer_test.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/testutils"
+	"github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegisterOperator(t *testing.T) {
```
Comment (afkbyte) [inline chainio/clients/elcontracts/writer_test.go:19]: similar comments for tests as in https://github.com/Layr-Labs/eigensdk-go/pull/360

would prefer test config + helper function for the anvil state setup

File: chainio/clients/builder.go (hunk 1 above)
Comment (TomasArrachea) [reply chainio/clients/builder.go]: The `RewardsCoordinatorAddress` is part of the [elcontracts.Config](https://github.com/Layr-Labs/eigensdk-go/blob/dev/chainio/clients/elcontracts/reader.go#L25), and its used when [building the ElContracts Clients](https://github.com/Layr-Labs/eigensdk-go/blob/dev/chainio/clients/elcontracts/builder.go). Anyways, its was needed here to test the rewards methods, but we decided not to add those tests in this PR, so I could remove this for now.

**Comment by afkbyte** [comment]
Let's bring in the changes from https://github.com/Layr-Labs/eigensdk-go/pull/360 then looks good to merge to me!

**Comment by TomasArrachea** [comment]
> Let's bring in the changes from #360 then looks good to merge to me!

@afkbyte Done!

**Review by afkbyte — APPROVED** [review]
lgtm!

**Pull request merged**